src/decisionengine/framework/util/tests/test_metrics.py

```diff
@@ -0,0 +1,150 @@
+import pytest
+import tempfile
+
+
+@pytest.fixture(autouse=True)
+def env_setup(monkeypatch):
+    """Make sure we have a directory set for PROMETHEUS_MULTIPROC_DIR so that
+    metric instantiation gives us multiprocess metrics"""
+    with tempfile.TemporaryDirectory() as d:
```

Comment:
I think `tempfile.TemporaryDirectory()` needs to be moved out of this function.  Otherwise you'll get a new directory name each time.  With prometheus setting the `MULTIPROC_DIR` at import time, I'm not clear having this inside of env_setup will achieve the intended result.

Reply:
My understanding was that at instantiation of the Gauge metric, the prometheus_client library decides which Value class to use (https://github.com/prometheus/client_python/blob/master/prometheus_client/values.py), based on whether PROMETHEUS_MULTIPROC_DIR was set in the environment. For some reason, when I was debugging both these tests and the actual code, that check happened not when I first instantiated Gauge but at import time. As in, the value class for all metrics was set at import time. I'm not sure why. 

Either way, I'll change it back and leave the `delenv` in there too.

Reply:
Figured it out.  On running `import prometheus_client`, the `prometheus_client` library has an `__init__.py` file that imports the `metrics` module (https://github.com/prometheus/client_python/blob/master/prometheus_client/__init__.py#L4), which in turn imports the `values` module (https://github.com/prometheus/client_python/blob/master/prometheus_client/metrics.py#L6).  `values.py` has at the very bottom, the `ValueClass` global variable assigned to the return value of `get_value_class()`.  This latter function checks the environment.  That's why even though it looks like the `PROMETHEUS_MULTIPROC_DIR` environment variable should be getting checked at __init__, it actually gets checked at import.  This class that gets assigned to `ValueClass` is then used subsequently for all metrics.

Reply:
That explains it!